src/reconcile/rec_visibility.c

```diff
@@ -437,28 +438,40 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
              * In this case, we should leave the selected update unset to indicate that we want to
              * keep the same on-disk value but set the stop time point to indicate that the validity
              * window ends when this tombstone started.
+             *
+             * FIXME-WT-6557: no need to check this after WT-6557 is done as the tombstone will be
+             * freed when it is written to the disk image in the previous eviction.
              */
-            WT_ERR(__rec_append_orig_value(session, page, tombstone, vpack));
+            if (!F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS)) {
```

Comment:
Since we are appending the value if either of these flags (WT_UPDATE_RESTORED_FROM_DS, WT_UPDATE_RESTORED_FROM_HS) is not set. Do we still have to check for these flags inside `__rec_append_orig_value` function ?

Reply:
Yes, the function is called in another place as well.

Reply:
Thanks @quchenhao